packages/framework-provider-azure/src/helpers/query-helper.ts

```diff
@@ -214,3 +218,83 @@ function toLocalSortFor(
   })
   return sortedList
 }
+
+function buildProjections(projections: ProjectionFor<unknown> | string = '*'): string {
+  if (typeof projections !== 'object') {
+    return projections
+  }
+
+  // Group fields by the root property
+  const groupedFields: any = {}
+  Object.values(projections).forEach((field: string) => {
+    const root: string = field.split('.')[0]
+    if (!groupedFields[root]) {
+      groupedFields[root] = []
+    }
+    groupedFields[root].push(field)
+  })
+
+  return Object.keys(groupedFields)
+    .map((root: string): string => {
+      const fields = groupedFields[root]
+      if (root.endsWith('[]')) {
+        const arrayRoot: string = root.slice(0, -2)
+        const subFields = fields.map((f: string) => f.replace(`${root}.`, 'item.')).join(', ')
+        return `ARRAY(SELECT ${subFields} FROM item IN c.${arrayRoot}) AS ${arrayRoot}`
+      } else if (fields.length === 1 && !fields[0].includes('.')) {
+        // Simple field
+        return `c.${fields[0]}`
+      } else {
+        // Nested object fields
+        return fields.map((f: string) => `c.${f} AS "${f}"`).join(', ')
```

Comment:
In the previous `else if` branch you checked if the field didn't contain dots (`.`). Shouldn't we also check here?
Another question: Why in this case we add an `AS` but not in the previous branch?

Reply:
@alvaroloes the previous `else if` condition was when there was a single field and that it didn't contain a dot, which translates into meaning that the field is a top-level field. The `AS` is added for the `else` branch because it indicates nested properties in the read model, e.g., `foo.bar` indicates `{"foo" : {"bar": "some value"}}`. The `AS` is used to retrieve the entire path in your DB query result, e.g., `{"foo.bar": "some value"}` and then the `nestProperties` method converts it back to its original structure. The reason for this is that if one queries without the `AS`, CosmosDB doesn't retain the object structure. In the example, querying for `foo.bar` without the `AS` would return `{"bar": "some value"}`

Reply:
Woah! Nice trick what you did with the AS there. Thanks!